### PR TITLE
Added Thingpulse ESP32 gateways

### PR DESCRIPTION
**Pull request by hallard**
## Description:

Replace #1324 

- Added Thingpulse ESP32 gateways
- Added some RGB LED options

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/1technophile/OpenMQTTGateway/blob/development/docs/participate/development.md#developer-certificate-of-origin).

### CODE REVIEW

File: main/User_config.h [hunk 1]
```diff
@@ -454,28 +454,42 @@ CRGB leds[FASTLED_IND_NUM_LEDS];
...
 #  define SetupIndicators()                                                               \
     pinMode(RGB_LED_POWER, OUTPUT);                                                       \
     digitalWrite(RGB_LED_POWER, HIGH);                                                    \
     FastLED.addLeds<FASTLED_IND_TYPE, FASTLED_IND_DATA_GPIO>(leds, FASTLED_IND_NUM_LEDS); \
-    FastLED.setBrightness(20);
-#  define ErrorIndicatorON() \
-    leds[0] = CRGB::Orange;  \
+    FastLED.setBrightness(FASTLED_BRIGHTNESS);
+#  define ErrorIndicatorON()             \
+    leds[FASTLED_ERROR_LED] = CRGB::Red; \
```
Comment (1technophile) [inline main/User_config.h]: Could you set it to Orange please, I have found the Red to be a bit violent for errors

File: platformio.ini [hunk 2]
```diff
@@ -890,6 +890,10 @@ build_flags =
   ${com-esp.build_flags}
   '-DZgatewayBT="BT"'
   '-DGateway_Name="OpenMQTTGateway_ESP32_TINYPICO_BLE"'
+  '-DRGB_LED_POWER=13'
+  '-DFASTLED_IND_DATA_GPIO=2,12,RGB'
+  '-DFASTLED_IND_TYPE=APA102'
+  '-DFASTLED_IND_NUM_LEDS=1'
```
Comment (1technophile) [inline platformio.ini]: Maybe this could be in a separate PR

**Comment by 1technophile** [comment]
Thanks, could you add the environment bvelow:
https://github.com/1technophile/OpenMQTTGateway/blob/827724f079b9cd6cee6ce69735aee5bb011dced5/.github/workflows/build.yml#L10

And also replace:
`    if 'firmware.bin' in name and ('esp32' in name or 'ttgo' in name or 'heltec' in name):`
by
`    if 'firmware.bin' in name and ('esp32' in name or 'ttgo' in name or 'heltec' in name or 'thingpulse' in name):`
Below:
https://github.com/1technophile/OpenMQTTGateway/blob/827724f079b9cd6cee6ce69735aee5bb011dced5/scripts/gen_wu.py#L122
and below:
https://github.com/1technophile/OpenMQTTGateway/blob/827724f079b9cd6cee6ce69735aee5bb011dced5/scripts/gen_wu_dev.py#L110

File: platformio.ini (hunk 2 above)
Comment (hallard) [reply platformio.ini]: yep

File: main/User_config.h (hunk 1 above)
Comment (hallard) [reply main/User_config.h]: Done

**Comment by hallard** [comment]
All done

**Comment by 1technophile** [comment]
Thanks
